Fixing README Setup Instructions

jsnorek opened the pull request:
This PR is for issue #463

It includes an update to the "Getting Started" section, including additional instructions to set up your virtual environment, installing dependencies, installing pre-commit hooks, and creating the config.ini file. 

![Screenshot 2025-01-15 at 2 37 46 PM](https://github.com/user-attachments/assets/51b1672b-b313-4188-af3f-e3b1a9e9c24c)

I've also added the Square Testing instructions from @monikkaelyse 
![Screenshot 2025-01-15 at 2 38 46 PM](https://github.com/user-attachments/assets/4202612e-a45f-4658-aff4-019ee3cb5a8d)

These additions were also noted in the table of contents at the beginning of the README
![Screenshot 2025-01-15 at 2 41 04 PM](https://github.com/user-attachments/assets/62d786f2-f246-4bd7-a3a9-7e273cbbe95e)

jsnorek commented:
Please note that I had an issue with Prettier when pushing my commits and they all went into one commit along with the Prettier formatting changes. I'm working to resolve that

ChasVanDav commented on .pre-commit-config.yaml:
```diff
@@ -38,5 +38,5 @@ repos:
       - id: prettier
 default_language_version:
-  python: python3.6
+  python: python3.8
```
Should we add the specific version since the dependencies are so picky? python3.8.10

jsnorek replied:
Yes, I've updated this!

ChasVanDav reviewed (changes requested):
I have one question about line 41, otherwise it looks clear and well-formatted.

jsnorek commented:
I had to rebase this branch this morning, so if anyone had pulled it for review, please delete and re-pull :)

jsnorek commented:
@monikkaelyse @daaimah123 this has been updated and is ready for review!

daaimah123 commented on README.md:
```diff
@@ -7,10 +7,17 @@ currently hosted on DreamHost.
 - [What](#what)
 - [How](#how)
 - [Getting Started](#getting-started)
+  - [Set Up Virtual Environment](#set-up-virtual-environment)
+  - [Install pip Version 23](#install-pip-version-23)
+  - [Install and Upgrade pip-tools](#install-and-upgrade-pip-tools)
   - [Install Pre-Commit Hooks](#install-pre-commit-hooks)
+  - [Install Requirements](#install-requirements)
+  - [Create Config.ini File](#create-configini-file)
   - [Running Locally](#running-locally)
   - [Using Docker to Run Locally](#using-docker-to-run-locally)
   - [CSS / SCSS](#css--scss)
+  - [Square Testing](#square-testing)
+  - [Updating Demographics Chart](#updating-the-demographics-chart-for-the-apply-section)
```
Great hyperlink additions

daaimah123 commented on README.md:
```diff
@@ -33,7 +40,41 @@ https://docs.google.com/document/d/1oL3BaemFfUD7DfoFzhTSwcX4lPxYbWN3Dy9oZFfGP0Y/
 ## Getting Started
 
-This app uses Python 3.6; please stick to this version when doing development.
+You need Python version 3.8.10 and pip version 23 in order to properly update dependencies.
```
Can you update [L38](https://github.com/Techtonica/techtonica.org/blob/develop/README.md?plain=1#L31) to hyperlink to the url in [L39](https://github.com/Techtonica/techtonica.org/blob/develop/README.md?plain=1#L32), and remove the colon and url?

> [Detailed instructions on how to update the website](https://docs.google.com/document/d/1oL3BaemFfUD7DfoFzhTSwcX4lPxYbWN3Dy9oZFfGP0Y/edit)

jsnorek replied:
I've added this change in this commit :) [2a509e1](https://github.com/Techtonica/techtonica.org/pull/482/commits/2a509e1ff06e0c907234e1345d7ea7942f5007d2)

ChasVanDav reviewed (approved):
Great job setting super clear local env set up instructions!

jsnorek commented on README.md:
```diff
@@ -110,50 +187,80 @@ sass static/sass/style.scss static/css/style.css
-1. Edit `requirement.in`
+1. Edit `requirements.in`
 1. Generate `requirements.txt`:
 
```
Are you asking if we should add `pip install -r dev.txt` to the command on line 266?

<img width="740" alt="Screenshot 2025-01-17 at 5 05 15 PM" src="https://github.com/user-attachments/assets/fda5d8b5-3eb3-48b7-be42-531aa05f8e9a" />

jsnorek commented on README.md:
````diff
@@ -74,7 +151,7 @@ Browse to <http://localhost:5000>.
+pip install pyopenssl
 FLASK_DEBUG=1 FLASK_APP=main_site.py FLASK_RUN_CERT=adhoc flask run
 ```
 
````
I get the same error - when I change the docker yaml file to port 5001 it works on docker but get this error on the localhost link 
![Screenshot 2025-01-17 at 5 51 43 PM](https://github.com/user-attachments/assets/83f3aebd-a712-49f8-81cb-b77ceca5bf6e)

jsnorek replied:
Confirmed Docker instructions are not working at the moment (but the Docker instructions in the README are technically accurate). Opened in a new [issue](https://github.com/Techtonica/techtonica.org/issues/510)

jsnorek commented on README.md:
```diff
@@ -110,50 +187,80 @@ sass static/sass/style.scss static/css/style.css
-1. Edit `requirement.in`
+1. Edit `requirements.in`
 1. Generate `requirements.txt`:
 
```
This has been added as an additional [issue ](https://github.com/Techtonica/techtonica.org/issues/511)

jsnorek commented on README.md:
```diff
@@ -43,21 +83,58 @@ consistency.
 
 ### Running Locally
 
 [If you prefer using Docker, see instructions](#using-docker-to-run-locally).
```
Added the anchor tag 👍 

Docker instructions error is referenced in another conversation, but another [issue](https://github.com/Techtonica/techtonica.org/issues/510) was opened to address this

jsnorek commented:
Addressed all comments - 
Confirmed Docker instructions are not working at the moment (but the Docker instructions in the README are technically accurate). Opened in a new [issue](https://github.com/Techtonica/techtonica.org/issues/510)
Another [issue ](https://github.com/Techtonica/techtonica.org/issues/511) has also been opened for the error with ``` pre-commit run seed-isort-config -a --hook-stage manual ```

jsnorek commented:
@monikkaelyse I added your changes 👍
